### PR TITLE
New format for the OCaml version string

### PR DESCRIPTION
**Pull request by Octachron**
This PR proposes to change the specification of the OCaml version string to allow "~" with the Debian semantic.
The new format is `major.minor[.patchlevel][(+|~)additional-info]`, for instance `4.12.0~beta1+flambda`.

This change will make possible to use Debian ordering for pre-release version (alpha, beta, and rcs)
```
4.12.0~alpha1 < 4.12.0 < 4.12.0+trunk
```
For now, this is only a documentation change, and we propose to postpone any use of the new format to 4.12 and later.

### CODE REVIEW

**Review by xavierleroy — APPROVED** [review]
Approving based on positive opinions expressed during a discussion among the core developers.

**Comment by damiendoligez** [comment]
Looks good. @Octachron don't forget to update the Changes entry before merging.

**Pull request merged**
